lib/smooth/core/io/i2c/I2CMasterDevice.cpp

```diff
@@ -217,27 +299,45 @@ namespace smooth::core::io::i2c
         i2c_reset_rx_fifo(port);
     }
 
-    void I2CMasterDevice::log_error(esp_err_t err, const char* msg)
+    // Log the error
+    //void I2CMasterDevice::log_error(esp_err_t err, const char* msg)
+    void I2CMasterDevice::log_error(esp_err_t err, uint8_t address)
     {
+        std::stringstream ss;
+        ss << "Error in - " << process_name << " - slave address: 0x" << std::hex << static_cast<int32_t>(address);
+
         if (err == ESP_ERR_INVALID_ARG)
         {
-            Log::error(log_tag, "{} - Parameter error", msg);
+            Log::error(log_tag, "{} - Parameter error", ss.str().c_str());
```

Comment:
Is c_str() really needed, doesn't fmt accept a string as an argument?

Reply:
Yes you are correct - removing c_str()